src/main/java/com/gznznzjsn/outstagram/service/impl/SubscriptionServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.gznznzjsn.outstagram.service.impl;
+
+import com.gznznzjsn.outstagram.model.exception.IllegalActionException;
+import com.gznznzjsn.outstagram.model.node.Account;
+import com.gznznzjsn.outstagram.model.relationship.Subscription;
+import com.gznznzjsn.outstagram.persistence.repository.SubscriptionRepository;
+import com.gznznzjsn.outstagram.service.SubscriptionService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class SubscriptionServiceImpl implements SubscriptionService {
+
+    private final SubscriptionRepository repository;
+
+    @Override
+    public void subscribe(final UUID sourceId, final UUID targetId) {
+        if (sourceId.equals(targetId)) {
+            throw new IllegalActionException("You can't subscribe to yourself");
+        }
+        var source = Account.builder()
```

Comment:
@gznznzjsn I don't recommend to use `var`, simply because it's not so traceable on a code review

Reply:
@h1alexbel  I use it only where the type is obvious (for example, with builders) in order to have more compact lines of code